File: JavaScript/1151-Minimum Swaps to Group All 1's Together.js
/**
 * @param {number[]} data
 * @return {number}
 */
var minSwaps = function(data) {
    let ones = data.reduce((acc, x)=>acc+x);
    if (ones === 0) return 0;
    
    let currOne = maxOne = data.slice(0, ones).reduce((acc,x)=>acc+x);
    
    for (let i=ones; i<data.length; i++){
        currOne += data[i] - data[i-ones];
        maxOne = Math.max(maxOne, currOne);
    }
    return ones - maxOne;
};

File: 1151-Minimum Swaps to Group All 1's Together.py
class Solution(object):
    def minSwaps(self, data):
        """
        :type data: List[int]
        :rtype: int
        """
        ones = sum(data)
        currOne = maxOne = sum(data[:ones])
        
        for right in range(ones, len(data)):
            currOne += data[right] - data[right-ones]
            maxOne = max(maxOne, currOne)
        
        return ones-maxOne